From f0965256da501f2787f325a38900933fc9ab312d Mon Sep 17 00:00:00 2001
From: Ma Jun <majun258@linux.alibaba.com>
Date: Tue, 25 Feb 2020 16:11:49 +0800
Subject: [PATCH] Update the opensbi version

[1]Update the sbi version from 0.3 to 0.6.
[2]Add the buidl command for c910

Signed-off-by: Ma Jun <majun258@linux.alibaba.com>
---
 boot/opensbi/Config.in  | 10 ++++++++
 boot/opensbi/opensbi.mk | 51 +++++++++++++++++++++++++++++++++--------
 2 files changed, 52 insertions(+), 9 deletions(-)

diff --git a/boot/opensbi/Config.in b/boot/opensbi/Config.in
index 5f3cc13..8a9f90a 100644
--- a/boot/opensbi/Config.in
+++ b/boot/opensbi/Config.in
@@ -22,4 +22,14 @@ config BR2_TARGET_OPENSBI_PLAT
 	  library libsbi.a is built. If a platform is specified then
 	  the platform specific static library libplatsbi.a and firmware
 	  examples are built.
+
+config BR2_TARGET_OPENSBI_C910_PLAT
+	string "CSKY C910 SBI Platform"
+	default "thead/c910"
+	help
+	  Specifies the OpenSBI platform to build. If no platform is
+	  specified only the OpenSBI platform independent static
+	  library libsbi.a is built. If a platform is specified then
+	  the platform specific static library libplatsbi.a and firmware
+	  examples are built.
 endif
diff --git a/boot/opensbi/opensbi.mk b/boot/opensbi/opensbi.mk
index 45a3fc4..f3ccfc8 100644
--- a/boot/opensbi/opensbi.mk
+++ b/boot/opensbi/opensbi.mk
@@ -4,30 +4,63 @@
 #
 ################################################################################
 
-OPENSBI_VERSION = 0.3
+OPENSBI_VERSION = 0.6
 OPENSBI_SITE = $(call github,riscv,opensbi,v$(OPENSBI_VERSION))
 OPENSBI_LICENSE = BSD-2-Clause
 OPENSBI_LICENSE_FILES = COPYING.BSD
 OPENSBI_INSTALL_TARGET = NO
 OPENSBI_INSTALL_STAGING = YES
 
-OPENSBI_MAKE_ENV = \
+OPENSBI_QEMU_MAKE_ENV = \
 	CROSS_COMPILE=$(TARGET_CROSS)
 
-OPENSBI_PLAT = $(call qstrip,$(BR2_TARGET_OPENSBI_PLAT))
-ifneq ($(OPENSBI_PLAT),)
-OPENSBI_MAKE_ENV += PLATFORM=$(OPENSBI_PLAT)
+OPENSBI_C910_MAKE_ENV = \
+	CROSS_COMPILE=$(TARGET_CROSS)
+
+OPENSBI_QEMU_PLAT = $(call qstrip,$(BR2_TARGET_OPENSBI_PLAT))
+OPENSBI_C910_PLAT = $(call qstrip,$(BR2_TARGET_OPENSBI_C910_PLAT))
+
+ifneq ($(OPENSBI_QEMU_PLAT),)
+OPENSBI_QEMU_MAKE_ENV += PLATFORM=$(OPENSBI_QEMU_PLAT)
+endif
+
+ifneq ($(OPENSBI_C910_PLAT),)
+OPENSBI_C910_MAKE_ENV += PLATFORM=$(OPENSBI_C910_PLAT)
 endif
 
 define OPENSBI_BUILD_CMDS
-	$(TARGET_MAKE_ENV) $(OPENSBI_MAKE_ENV) $(MAKE) -C $(@D)
+	$(TARGET_MAKE_ENV) $(OPENSBI_C910_MAKE_ENV) \
+	FW_TEXT_START=0xc0000000 \
+	FW_JUMP_ADDR=0xc0200000 \
+	$(MAKE) -C $(@D)
+	mv $(@D)/build/platform/$(OPENSBI_C910_PLAT)/firmware/fw_jump.elf $(@D)/fw_jump_3G.elf
+	mv $(@D)/build/platform/$(OPENSBI_C910_PLAT)/firmware/fw_jump.bin $(@D)/fw_jump_3G.bin
+	mv $(@D)/fw_jump*.elf $(BINARIES_DIR)/hw/
+	mv $(@D)/fw_jump*.bin $(BINARIES_DIR)/hw/
+
+	$(MAKE) -C $(@D) distclean
+	$(MAKE) -C $(@D) clean
+
+	$(TARGET_MAKE_ENV) $(OPENSBI_C910_MAKE_ENV) \
+	FW_TEXT_START=0x00000000 \
+	FW_JUMP_ADDR=0x00200000 \
+	$(MAKE) -C $(@D)
+	mv $(@D)/build/platform/$(OPENSBI_C910_PLAT)/firmware/fw_jump.elf $(@D)/fw_jump_0G.elf
+	mv $(@D)/build/platform/$(OPENSBI_C910_PLAT)/firmware/fw_jump.bin $(@D)/fw_jump_0G.bin
+	mv $(@D)/fw_jump*.elf $(BINARIES_DIR)/hw/
+	mv $(@D)/fw_jump*.bin $(BINARIES_DIR)/hw/
+	
+	$(MAKE) -C $(@D) distclean
+	$(MAKE) -C $(@D) clean
+
+	$(TARGET_MAKE_ENV) $(OPENSBI_QEMU_MAKE_ENV) $(MAKE) -C $(@D)
 endef
 
-ifneq ($(OPENSBI_PLAT),)
+ifneq ($(OPENSBI_QEMU_PLAT),)
 OPENSBI_INSTALL_IMAGES = YES
 define OPENSBI_INSTALL_IMAGES_CMDS
-	$(INSTALL) -m 0644 -D $(@D)/build/platform/$(OPENSBI_PLAT)/firmware/fw_jump.bin $(BINARIES_DIR)/fw_jump.bin
-	$(INSTALL) -m 0644 -D $(@D)/build/platform/$(OPENSBI_PLAT)/firmware/fw_jump.elf $(BINARIES_DIR)/fw_jump.elf
+	$(INSTALL) -m 0644 -D $(@D)/build/platform/$(OPENSBI_QEMU_PLAT)/firmware/fw_jump.bin $(BINARIES_DIR)/fw_jump.bin
+	$(INSTALL) -m 0644 -D $(@D)/build/platform/$(OPENSBI_QEMU_PLAT)/firmware/fw_jump.elf $(BINARIES_DIR)/fw_jump.elf
 endef
 endif
 
-- 
2.17.1

